hera_mc/geo_handling.py

```diff
@@ -19,298 +21,397 @@
+        self.session = db.sessionmaker()
+        self.station_types = None
+
+    def close(self):
+        self.session.close()
+
+    def get_station_types(self,add_stations=True):
+        """
+        adds a dictionary of sub-arrays (station_types) to the class
+             [prefix]{'Description':'...', 'plot_marker':'...', 'stations':[]}
+        also adds the "flipped" dictionary
+
+        return dictionary with station information
+
+        Parameters:
+        -------------
+        add_stations:  if True, add all of the stations to their types
+                       if False, just return station types
+        """
+
+        station_data = self.session.query(geo_location.StationType).all()
+        stations = {}
+        for sta in station_data:
+            stations[sta.prefix] = {'Name': sta.station_type_name,
+                                    'Description': sta.description,
+                                    'Marker': sta.plot_marker, 'Stations': []}
+        if add_stations:
+            locations = self.session.query(geo_location.GeoLocation).all()
+            for loc in locations:
+                for k in stations.keys():
+                    if loc.station_name[:len(k)] == k:
+                        stations[k]['Stations'].append(loc.station_name)
+        self.station_types = stations
+        self.flipped_station_types = {}
+        for key in self.station_types.keys():
+            self.flipped_station_types[self.station_types[key]['Name']] = key
+
+    def is_in_geo_location(self, station_name):
+        """
+        checks to see if a station_name is in the geo_location database
+
+        return True/False
+
+        Parameters:
+        ------------
+        station_name:  string name of station
+        """
+
+        station = self.session.query(geo_location.GeoLocation).filter(geo_location.GeoLocation.station_name == station_name)
+        if station.count() > 0:
+            station_present = True
         else:
-            raise ValueError('More than one active connection')
-    return antenna_connected
+            station_present = False
+        return station_present
 
 
-def locate_station(args, to_find, query_date, show_location=False):
-    """
-    Return the location of station_name or antenna_number as contained in args.locate.
-    This accepts the fact that antennas are sort of stations, even though they are parts
+    def is_in_connections(self, station_name, active_date=None):
+        """
+        checks to see if the station_name is in the connections database (which means it is also in parts)
 
-    Parameters:
-    ------------
-    args:  needed arguments to open database
-    station_to_fine:  station name to find
-    query_date:  astropy Time for contemporary antenna
-    show_location:   if True, it will print the information.
-    """
+        return True/False unless check_if_active flag is set, when it returns the antenna number at that location
```

Comment:
Is the check_if_active flag meant to be passed into this method? If not, and since it doesn't appear to be an attribute on the class, the output of the method should not depend on it.

Actually, this documentation appears to be wrong because I don't see this flag used anywhere in the function.